test/it/ChoiceSpec.scala

```diff
@@ -29,12 +29,12 @@ class ChoiceSpec extends IntegrationSpec {
       status(response) mustBe FORBIDDEN
     }
 
-    "return 200" in {
+    "return 303" in {
       givenAuthSuccess()
 
       val response = get(controllers.routes.ChoiceController.displayPage())
 
-      status(response) mustBe OK
+      status(response) mustBe SEE_OTHER
```

Comment:
This is not a full solution, because now you have 2 cases: present the choice page if there is cache, or redirect if there is not.
Please add a test case to cover both scenarios.